UM/View/SelectionPass.py

```diff
@@ -36,22 +42,34 @@ def __init__(self, width, height):
 
         self._selection_map = {}
         self._toolhandle_selection_map = {
-            Color.dropLowBits(self._dropAlpha(ToolHandle.DisabledSelectionColor)): ToolHandle.NoAxis,
-            Color.dropLowBits(self._dropAlpha(ToolHandle.XAxisSelectionColor)): ToolHandle.XAxis,
-            Color.dropLowBits(self._dropAlpha(ToolHandle.YAxisSelectionColor)): ToolHandle.YAxis,
-            Color.dropLowBits(self._dropAlpha(ToolHandle.ZAxisSelectionColor)): ToolHandle.ZAxis,
-            Color.dropLowBits(self._dropAlpha(ToolHandle.AllAxisSelectionColor)): ToolHandle.AllAxis,
-            Color.dropLowBits(ToolHandle.DisabledSelectionColor): ToolHandle.NoAxis,
-            Color.dropLowBits(ToolHandle.XAxisSelectionColor): ToolHandle.XAxis,
-            Color.dropLowBits(ToolHandle.YAxisSelectionColor): ToolHandle.YAxis,
-            Color.dropLowBits(ToolHandle.ZAxisSelectionColor): ToolHandle.ZAxis,
-            Color.dropLowBits(ToolHandle.AllAxisSelectionColor): ToolHandle.AllAxis
+            self._dropAlpha(ToolHandle.DisabledSelectionColor): ToolHandle.NoAxis,
+            self._dropAlpha(ToolHandle.XAxisSelectionColor): ToolHandle.XAxis,
+            self._dropAlpha(ToolHandle.YAxisSelectionColor): ToolHandle.YAxis,
+            self._dropAlpha(ToolHandle.ZAxisSelectionColor): ToolHandle.ZAxis,
+            self._dropAlpha(ToolHandle.AllAxisSelectionColor): ToolHandle.AllAxis,
+            ToolHandle.DisabledSelectionColor: ToolHandle.NoAxis,
+            ToolHandle.XAxisSelectionColor: ToolHandle.XAxis,
+            ToolHandle.YAxisSelectionColor: ToolHandle.YAxis,
+            ToolHandle.ZAxisSelectionColor: ToolHandle.ZAxis,
+            ToolHandle.AllAxisSelectionColor: ToolHandle.AllAxis
         }
 
+        self._mode = SelectionPass.SelectionMode.OBJECTS
+        Selection.selectedFaceChanged.connect(self._onSelectedFaceChanged)
+
         self._output = None
 
+    def _onSelectedFaceChanged(self):
+        self._mode = SelectionPass.SelectionMode.FACES if Selection.getFaceSelectMode() else SelectionPass.SelectionMode.OBJECTS
+
     ##  Perform the actual rendering.
     def render(self):
+        if self._mode == SelectionPass.SelectionMode.OBJECTS:
+            self._renderObjectsMode()
+        elif self._mode == SelectionPass.SelectionMode.FACES:
+            self._renderFacesMode()
```

Comment:
I'd add an entry for `else` to log as a warning in case the mode doesn't match any of the above.